### PR TITLE
OLS-281: Implements azure openai support

### PR DESCRIPTION
**Pull request by thoraxe**
## Description
This PR implements Azure OpenAI support using Langchain's `AzureChatOpenAI`

## Type of change

- [ ] Refactor
- [X] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [X] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents
[OLS-53](https://issues.redhat.com//browse/OLS-53)
[OLS-281](https://issues.redhat.com//browse/OLS-281)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [X] If it is a core feature, I have added thorough tests.

## Testing
Added tests for:
```
tests/config/singleprovider.e2e.template.config.yaml
tests/unit/app/models/test_config.py
tests/unit/llms/providers/test_azure_openai.py
```

Although it seems like the e2e needs to be changed because of the requirements on a URL that Azure OpenAI brings.

### CODE REVIEW

File: ols/src/llms/providers/azure_openai.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            "temperature": 0.01,
+            "max_tokens": 512,
+            "top_p": 0.95,
+            "frequency_penalty": 1.03,
+            "verbose": False,
+        }
+
+    def load(self) -> LLM:
+        # TODO: these are not allowed for OpenAI - we need to figure
+        # out first how to handle params in general - whitelist/validation, ...
```
Comment (bparees) [inline ols/src/llms/providers/azure_openai.py:45]: you can ref this story: https://issues.redhat.com/browse/OLS-363

Comment (thoraxe) [reply ols/src/llms/providers/azure_openai.py:45]: This was added.

**Comment by bparees** [comment]
@thoraxe verify job is failing

File: ols/app/models/config.py [hunk 2]
```diff
@@ -183,6 +184,13 @@ def __init__(self, data: Optional[dict] = None) -> None:
                 raise InvalidConfigurationError("model name is missing")
             model = ModelConfig(m)
             self.models[m["name"]] = model
+        if self.type == constants.PROVIDER_AZURE_OPENAI:
+            # deployment_name only required when using Azure OpenAI
+            self.deployment_name = data.get("deployment_name", None)
+            if self.deployment_name is None:
+                raise InvalidConfigurationError(
+                    f"deployment_name is required for Azure OpenAI provider {self.name}"
```
Comment (bparees) [inline ols/app/models/config.py:192]: we need a test that ensures this is enforced, especially so when @m-danilow refactors our validation logic, we've got a test that enforces the expected behavior.

i think that is the last thing needed here.  we're also going to have to work out how to have configs for azureai that specify the right deployment name value, but we can handle that in a followup refactoring of how e2e config is managed.

**Pull request merged**